dot/network/service.go

```diff
@@ -321,6 +327,35 @@ main:
 			if err != nil {
 				logger.Debug("problem sending system.interval telemetry message", "error", err)
 			}
+			netState := make(map[string]interface{})
+			netState["peerId"] = s.host.h.ID()
+			hostAddrs := []string{}
+			for _, v := range s.host.h.Addrs() {
+				hostAddrs = append(hostAddrs, v.String())
+			}
+			netState["externalAddressess"] = hostAddrs
+			listAddrs := []string{}
+			for _, v := range s.host.h.Network().ListenAddresses() {
+				listAddrs = append(listAddrs, v.String())
+			}
+			netState["listenedAddressess"] = listAddrs
+
+			peers := make(map[string]interface{})
+			for _, v := range s.Peers() {
+				p := &peerInfo{
+					Roles:      v.Roles,
+					BestHash:   v.BestHash.String(),
+					BestNumber: v.BestNumber,
+				}
+				peers[v.PeerID] = *p
+			}
+			netState["connectedPeers"] = peers
+			err = telemetry.GetInstance().SendMessage(telemetry.NewTelemetryMessage(
```

Comment:
After looking at the `msgToBytes` method it seems all that does is `json.Marshal()` the `map[string]interface{}`.  So there's no real need to construct these messages as `map[string]interface{}`.  You could essentially just pass in structs.  So maybe in this case: 

```
type netState struct {
    PeerID string `json:"peerId"`
    ExternalAddresses []string `json:"externalAddresses"`
    ListenedAddresses []string `json:"listenedAddresses"`  
}
```
You could create a constructor for this type as well to handle the population of the fields.